client/network/src/protocol/generic_proto/handler/group.rs

```diff
@@ -483,35 +483,35 @@ impl ProtocolsHandler for NotifsHandler {
 	) -> Poll<
 		ProtocolsHandlerEvent<Self::OutboundProtocol, Self::OutboundOpenInfo, Self::OutEvent, Self::Error>
 	> {
-		while let Poll::Ready(ev) = self.legacy.poll(cx) {
-			match ev {
+		if let Poll::Ready(ev) = self.legacy.poll(cx) {
```

Comment:
Was this change on purpose?

Reply:
Yes. It doesn't make sense to use a loop considering that all the code paths after entering the loop lead to a `return`. I moved the `return` right below the `if` to make that obvious.